### PR TITLE
Prevent installation of cantera from PyPI

### LINKED ISSUE #30: Conda path replacement mangling Cantera data directories
The behavior used by conda to replace paths on the build system with paths corresponding to the installation location is causing some problems with the Cantera data directories. Specifically, from the Python interface:
```py
>>> import cantera as ct
>>> ct.get_data_directories()
['.',
 '/home/speth/miniconda3/envs/ct26/lib/python3.10/site-packages/cantera/data',
 '/home/speth/miniconda3/envs/ct26/share/cantera/data\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
]
>>> ct.Solution('spam.yaml')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "build/python/cantera/base.pyx", line 71, in cantera._cantera._SolutionBase.__cinit__
  File "build/python/cantera/base.pyx", line 128, in cantera._cantera._SolutionBase._cinit
  File "build/python/cantera/base.pyx", line 215, in cantera._cantera._SolutionBase._init_yaml
cantera._cantera.CanteraError: 
*******************************************************************************
CanteraError thrown by Application::findInputFile:

Input file spam.yaml not found in directories 
'.', 
'/home/speth/miniconda3/envs/ct26/lib/python3.10/site-packages/cantera/data', 
'/home/speth/miniconda3/envs/ct26/share/cantera/data
>>>
```
The problem is that the last directory name is padded with null bytes. When this is inserted into the error message printed by `findInputFile`, the message apparently ends up being treated as a null-terminated string, and gets truncated there.

This problem also affects the conda-forge packages.

Version tested: `cantera                   2.6.0           py310h6cd0baa_0`

### LINKED ISSUE #31: `Conda Install` Development Cantera Version 3.0.0a2
**Problem description**

**Steps to reproduce**

I'm trying to install the Cantera development version 3.0.0a2 using `conda`. This is because I want to test the preconditioner functionality 

I use the following command to create a new `conda` environment and install the latest dev Cantera version (note that I locally removed the stable Cantera version from my machine) :

``` shell
conda create --name ct-dev --channel cantera/label/dev cantera ipython matplotlib jupyter
```
Using `ipython` I check the version of Cantera installed

```python
import cantera as ct
print(ct.__version__)
# 2.6.0
```
But version 2.6.0 gets printed

Using `conda list` command I can see that version 3.0.0a2 is installed. But, doing `pip list` shows that there is also the 2.6.0 version installed even though I never installed it. It seems to get installed automatically when I install the dev version 3.0.0a2 using `conda`

Is there any way to fix this? Am I doing something wrong?

Thank you in advance for the help!

**Behavior**

**System information**

- Cantera version: 3.0.0a2
- OS: Centos7 Linux
-  Architecture: x86
- Python Version: 3.9.12
- Python/MATLAB/other software versions: Python

**Attachments**

**Additional context**

### PR DESCRIPTION
**Pull request by ischoegl**
When running `pip install` in `build.sh/.bat`, fetching `cantera` from PyPi needs to be prevented.

Closes #31

Closes #30 (via Cantera/cantera#1358)
Circumvents #29 (MKL is disabled)

Output with [manually built package](https://gist.github.com/ischoegl/b7c935bb3c94403f3f4d4e5db67ac116):
```
>>> import cantera as ct
>>> ct.__version__
'3.0.0a2'
```

Other fixes:
- Include `ck2yaml` in package
- Update `pytest`
- Update deprecated SCons flag
- Temporarily disable MKL, see #29
- Add 'official' `share/cantera/data` to `libcantera` package

### CODE REVIEW

**Comment by ischoegl** [comment]
@bryanwweber ... regarding `pytest`, this can now be run as `pytest -vv path/to/samples/python`, but I'm not sure what folder `meta.yaml` runs in.

**Comment by bryanwweber** [comment]
NumPy pulls in MKL, so that's not totally surprising. I think we'd like to link Cantera to MKL as well, but we can fix that in a separate PR. Thanks for taking this on!

**Review by speth — APPROVED** [review]
Ah, nice job solving the mystery of how this could result in having Cantera 2.6.0 installed. I guess I shouldn't have been surprised that it was a result of `pip` doing something wrong.

**Comment by ischoegl** [comment]
@bryanwweber ... let me know if you have anything else to add? Won't merge this myself as this is the first time I'm working on this repo.

File: cantera/meta.yaml [hunk 1]
```diff
@@ -116,15 +117,16 @@ outputs:
     test:
       requires:
         - pytest
-      imports:
-        - cantera
```
Comment (bryanwweber) [inline cantera/meta.yaml:120]: I don't see a reason to remove this? It's just a very basic check that the package can be imported

File: cantera/build.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ echo "boost_inc_dir = '${PREFIX}/include'" >> cantera.conf
 if [[ "${OSX_ARCH}" == "" ]]; then
     echo "CC = '${CC}'" >> cantera.conf
     echo "CXX = '${CXX}'" >> cantera.conf
-    echo "blas_lapack_libs = 'mkl_rt,dl'" >> cantera.conf
+    # echo "blas_lapack_libs = 'mkl_rt,dl'" >> cantera.conf
```
Comment (bryanwweber) [inline cantera/build.sh:14]: Can you add a todo comment here and elsewhere in meta.yaml to note this temporary and that it's commented out to solve #31?

**Review by bryanwweber — CHANGES_REQUESTED** [review]
Thanks, a couple of tiny things here

**Comment by ischoegl** [comment]
@bryanwweber ... thanks! Comments should be taken care of.

**Review by bryanwweber — APPROVED** [review]

**Pull request merged**